src/main/kotlin/no/nav/familie/ef/sak/vilkår/gjenbruk/GjenbrukVilkårService.kt

```diff
@@ -50,6 +50,40 @@ class GjenbrukVilkårService(
         behandlingSomSkalOppdateres: UUID,
         behandlingIdSomSkalGjenbrukeInngangsvilkår: UUID,
     ) {
+        val vilkårsVurderingerForGjenbruk =
+            utledVilkårsvurderingerForGjenbrukData(
+                behandlingSomSkalOppdateres,
+                behandlingIdSomSkalGjenbrukeInngangsvilkår,
+            )
+        secureLogger.info(
+            "${SikkerhetContext.hentSaksbehandlerEllerSystembruker()} gjenbruker vurderinger fra behandling $behandlingIdSomSkalGjenbrukeInngangsvilkår " +
+                "for å oppdatere vurderinger på inngangsvilkår for behandling $behandlingSomSkalOppdateres",
+        )
+        vilkårsvurderingRepository.updateAll(vilkårsVurderingerForGjenbruk)
+    }
+
+    @Transactional
+    fun gjenbrukInngangsvilkårVurdering(
+        behandlingId: UUID,
+        behandlingIdForGjenbruk: UUID,
+        vilkårId: UUID,
+    ): Vilkårsvurdering {
+        val vilkårsVurderingForGjenbruk =
+            utledVilkårsvurderingerForGjenbrukData(
+                behandlingId,
+                behandlingIdForGjenbruk,
+            ).first { it.id == vilkårId }
+        secureLogger.info(
+            "${SikkerhetContext.hentSaksbehandlerEllerSystembruker()} gjenbruker enkel vurdering fra behandling $behandlingIdForGjenbruk " +
+                "for å oppdatere vurderinger på inngangsvilkår for behandling $behandlingId",
+        )
+        return vilkårsvurderingRepository.update(vilkårsVurderingForGjenbruk)
+    }
+
+    fun utledVilkårsvurderingerForGjenbrukData(
+        behandlingSomSkalOppdateres: UUID,
+        behandlingIdSomSkalGjenbrukeInngangsvilkår: UUID,
+    ): List<Vilkårsvurdering> {
         validerBehandlingForGjenbruk(
```

Comment:
Denne vil jo kaste feil i noen tilfeller nå - hvordan blir det når den brukes i en funksjon for å utlede vilkårsvurderinger? 
Før ble den brukt som sikkerhet for å unngå at vi lagret ned ugyldige vilkårsvurderinger...

Reply:
Nå tror jeg det kan komme situasjoner hvor ting krasjer ved uthenting av noe

Reply:
Denne valideringen må nok brukes når vi faktisk skal oppdatere data ved gjenbruk